### ISSUE TITLE
AntiDupl.NET-2.3.9

### ISSUE THREAD

**Issue by Men4a** [ermig1979/AntiDupl#19]
Новая версия падает через пару секунд после запуска, даже не начав сканирование. Предыдущая версия программы «AntiDupl.NET-2.3.8» работала отлично (редко , правда падала на битых файлах).

**Comment by heinrichI** [comment]
Выяснили что на некоторых процессорах не работает. Из-за нового компилятора.

**Comment by ermig1979** [comment]
Можно пересобрать со старой версией. Вопрос: с какой?

**Comment by Men4a** [comment]
AMD Phenom x2, разблоченный до х4, если это как-то поможет.

**Comment by ermig1979** [comment]
Пересобрал с использованием 2015 студии:
[AntiDupl.NET-2.3.10.zip](https://github.com/ermig1979/AntiDupl/files/1840451/AntiDupl.NET-2.3.10.zip)
Можете протестировать?

**Comment by ermig1979** [comment]
Пересобрал с использованием 2015 студии:
[AntiDupl.NET-2.3.10.zip](https://github.com/ermig1979/AntiDupl/files/1840451/AntiDupl.NET-2.3.10.zip)
Можете протестировать?

**Comment by Men4a** [comment]
Nope. При начале сканирования выскакивает прогресс-бар и пропадает. Выскакивает окно о прекращении работы. [картинко](https://img-fotki.yandex.ru/get/1032407/145751587.2cf/0_1e817e_ee27a18f_orig)

**Comment by ermig1979** [comment]
Версия без поддержки AVX и AVX2.
[AntiDupl.NET-2.3.10b.zip](https://github.com/ermig1979/AntiDupl/files/1840820/AntiDupl.NET-2.3.10b.zip)

**Comment by Men4a** [comment]
Увы. Вроде, начинает что-то искать, но через пару секунд выскакивает окошко с ошибкой. Проверил ещё раз версию «AntiDupl.NET-2.3.8» работает без проблем.

**Comment by ermig1979** [comment]
Версия, собранная 2013 студией.
[AntiDupl.NET-2.3.10c.zip](https://github.com/ermig1979/AntiDupl/files/1841137/AntiDupl.NET-2.3.10c.zip)

**Comment by valsaven** [comment]
AMD Phenom(tm) II X6 1090T (Windows 10 Pro x64) here. All three builds don't work :(

**Comment by ermig1979** [comment]
Т.е. можно исключить проблемы с новой версией студии и новыми инструкциями.

**Comment by ermig1979** [comment]
Т.е. можно исключить проблемы с новой версией студии и новыми инструкциями.

**Comment by ermig1979** [comment]
Т.е. можно исключить проблемы с новой версией студии и новыми инструкциями.

**Comment by ermig1979** [comment]
Т.е. можно исключить проблемы с новой версией студии и новыми инструкциями.

**Comment by ermig1979** [comment]
Т.е. можно исключить проблемы с новой версией студии и новыми инструкциями.

**Comment by heinrichI** [comment]
Почему? Если собранная в 2008 работает, так дело в студии.

**Comment by Men4a** [comment]
Версия «AntiDupl.NET-2.3.10c» сборки 2013 стидии ведёт себя аналогично остальным.

**Comment by Men4a** [comment]
Ура! Я смог сломать «AntiDupl.NET-2.3.8»: Снёс из «Компонентов Windows» предустановленнный .Net Framework 3.5. Скачал с сайта Microsoft online-установщик .Net Framework 3.5 и установил. В итоге обе версии: и «AntiDupl.NET-2.3.8» и «AntiDupl.NET-2.3.9» начали вылетать через 5 секунд после начала работы. 

Потом в «Компонентах Windows» включил .Net Framework 3.5, который установился, но обе программы работать дольше 10 секунд отказывались. Помогло только восстановление через консоль восстановления компонента из образа «Windows» .Net Framework 3.5. После этого «AntiDupl.NET-2.3.8» опять заработала.

Разработчики, признавайтесь, какие ещё примочки у вас на машинах стоят? И что за вкладка новая в конце в настройках появилась?

Кстати, через «Procmon.exe» отследил, что обе версии программы перед вылетом обращаются к какой-то .dll .NET Framework.

**Comment by ermig1979** [comment]
Framework 3.5 Дмитрий Борисов добавлял. Я вообще сидел на втором (по причине поддержки старых компьютеров).

**Comment by PalmtopTiger** [comment]
@ermig1979, возможно для проверки собрать x64 с поддержкой максимум SSE3?

**Comment by ermig1979** [comment]
Уже собирал  - не помогает.

**Comment by PalmtopTiger** [comment]
Не поделитесь ссылкой?
Если вы имеете в виду версию без поддержки AVX и AVX2, то, кроме этих инструкций, Феномы не поддерживают также SSSE3, SSE4.1 и SSE4.2.
Прошу прощения за назойливость, просто у меня совершенно нет идей, в чём ещё может быть дело.

**Comment by ermig1979** [comment]
Не в инструкциях дело. Библиотека Simd в любом случае перед использованием проверят наличие данного расширения CPU в runtime.

**Comment by heinrichI** [comment]
Значит осталось проверить собранных под 2010 и 2012. Все выше уже проверяли.

**Comment by ermig1979** [comment]
Убрал зависимость от .NET 3.5. Теперь используется .NET 2.0. Может кто сможет протестировать на старых машинах?

**Comment by KoalaBear84** [comment]
No one should be bothered with .NET 3.5 I guess, it even comes default on Windows 7, which reaches end-of-life in 2 months.

This issue can also be closed?

**Comment by ermig1979** [comment]
Yes it can be closed.
